### DISCUSSION TITLE
Global Rocket Instance?

### DISCUSSION THREAD

**Discussion by open-schnick** [rwf2/Rocket#2277] [Questions] [answered]
Hi basically as a follow up to [this](https://github.com/SergioBenitez/Rocket/discussions/2266) i have the question whether or not it is possible to have some kind of static instance of rocket.
Technically the instance should exist as long as the server is running right?
How do i get a static reference to the server or make it a global variable?

Basicly i am trying to do something like this:
```rust
rocket::build()
    .attach(AdHoc::on_ignite("some async messaging", |rocket| async {
        start_messaging(&rocket).await;
        rocket
    }))
    .mount("/", routes![index])
```

obiously i am getting:

```rust
35 |             start_messaging(&rocket).await;
   |             ----------------^^^^^^^-
   |             |               |
   |             |               borrowed value does not live long enough
   |             argument requires that `rocket` is borrowed for `'static`
36 |             rocket
37 |         }))
   |         - `rocket` dropped here while still borrowed
```

**Comment by the10thWiz** [comment, marked as answer]
Rocket isn't static, because it doesn't have to live as long as the program - if you use the launch attribute it will, but that's not required.

There are a few options here, depending on what you actually need. If you need to access some values managed by Rocket, you will have to wrap them in `Arc` and clone a reference to them.
